### ISSUE TITLE
Implementing Multi-select form fields; Formsnap doesn't appear to recognize Zod properties that are arrays

### ISSUE THREAD

**Issue by pheuter** [huntabyte/formsnap#27]
I'm trying to implement a MultiSelect form field to allow a user to select zero or many languages from a list of options:

```ts
languages: z.array(z.enum(["English", "Spanish", ...])),
```

But the Formsnap config appears to not recognize `languages` as a possible option for a form field:

```svelte
<Form.Field {config} name="languages">
  ...
</Form.Field>
```

```
Type '"languages"' is not assignable to type 'FormFieldName<ZodObject<{ firstName: ZodString; lastName: ZodString; preferredName: ZodNullable<ZodString>; ...>>'
```

In case I'm going about it the wrong way, is there a recommended approach for implementing multi-select form fields?

**Comment by huntabyte** [comment]
This is one of the trickier things when it comes to arrays with primitive values that I'm still trying to work out.

The issue though is that we're using `formFieldProxy`s, If you check [this part (https://superforms.rocks/components#using-the-componentized-field-in-awesome-ways) of the Superforms docs, you'll see that setup while okay for a tags input or something of that sort, isn't ideal for a multi-select, as it would expect each "option" to be a unique input, which is no good.

The maintainer of Superforms is currently on vacation, but once he's back I'm going to talk to him about some possible avenues to handle this, as right now when you use an `array` in the zod object, it wants the name to be `name[index]`.

I'm confident we'll be able to come up with a solution for this soon, even if I have to hack it!

**Comment by huntabyte** [comment]
Just providing an update here, I'm currently working on some ideas that @ciscoheat proposed that could accomplish this!

**Comment by BrandonTing** [comment]
@pheuter What's your work around for now? Thanks for sharing.

**Comment by pheuter** [comment]
Just don't use Formsnap for now

**Comment by huntabyte** [comment]
Hoping to  come to a clean solution to this soon!

**Comment by huntabyte** [comment]
Closed by #131
